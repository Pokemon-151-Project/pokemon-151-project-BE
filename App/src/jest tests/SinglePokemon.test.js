import SinglePokemon from "../components/SinglePokemon";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

//fake array with one pokemon for testing purposes
const fakePokemon = [
  {
    name: "Fake Pokemon Name",
    type: ["Grass", "Poison"],
    dexEntry: "blah blah blah",
    height: "4m",
    weight: "50kg",
  },
];

test("[1] renders without errors", () => {
  render(<SinglePokemon poke={fakePokemon[0]} />);
});

test("[2] Displays correct info on screen", () => {
  render(<SinglePokemon poke={fakePokemon[0]} />);

  const name = screen.queryByText(/fake pokemon name/i);
  const type = screen.queryByText(/grass/i);
  const dexEntry = screen.queryByText(/blah blah blah/i);
  const height = screen.queryByText(/4m/i);
  const weight = screen.queryByText(/50kg/i);
  expect(name).toBeVisible();
  expect(type).toBeVisible();
  expect(dexEntry).toBeVisible();
  expect(height).toBeVisible();
  expect(weight).toBeVisible();
});

test("[3] Toggles shiny image based on props", () => {
  let isShiny = false;

  const { rerender } = render(
    <SinglePokemon poke={fakePokemon[0]} isShiny={isShiny} />
  );

  let normalImg = screen.getByTestId("not-shiny");
  expect(normalImg).toBeVisible();

  isShiny = true;

  rerender(<SinglePokemon poke={fakePokemon[0]} isShiny={isShiny} />);
  let shinyImg = screen.getByTestId("shiny");
  expect(shinyImg).toBeVisible();
});

test("[4] Matches snapshot from 5.8.22", () => {
  const component = renderer.create(
    <SinglePokemon poke={fakePokemon[0]} isShiny={true} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
