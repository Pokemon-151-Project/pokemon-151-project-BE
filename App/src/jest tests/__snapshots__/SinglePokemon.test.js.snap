// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[4] Matches snapshot from 5.8.22 1`] = `
<article
  className="pokemon-card"
  data-testid="pokemon-card"
>
  <h2
    className="poke-name card-item"
  >
    Fake Pokemon Name
     
    <span
      className="dex-id"
    >
      #
    </span>
  </h2>
  <h3
    className="poke-type card-item"
  >
    Grass
     / 
    Poison
     
  </h3>
  <h4
    className="height-weight card-item"
  >
    Height: 
    4m
     
    <br />
    Weight: 
    50kg
  </h4>
  <figure
    className="img-container card-item"
  >
    <img
      alt="Fake Pokemon Name"
      className="poke-img"
      data-testid="shiny"
      src="https://www.serebii.net/Shiny/SWSH/undefined.png"
    />
  </figure>
  <blockquote
    className="card-item"
  >
    blah blah blah
  </blockquote>
</article>
`;
